File: Javascript/DaySix/CheckoutSystem.js
let  prompt = require('prompt-sync')();

let invoice = {}
let receipt = {}

function makeInvoice(product,price){
	if(price > 0)invoice[product] = price
	return invoice;
}

function makeReceipt(product,price){
	if(price > 0) receipt[product] = price
	return receipt;
}

function getSubTotal(sum, price) {
	if(price > 0) sum += price
	return sum;
}

function getVAT(sum) {
	let vat = sum * 0.075;
	return vat;
}

function getTotal(sum,vat){
	let total = sum + vat;
	return total;
}

function editInvoice(sum, vat, total, invoice){
	invoice["Subtotal"] = sum
	invoice["VAT"] = vat
	invoice["Total Amount"] = total
	return invoice;
}

function checkPayment(payment,total) {
	if(payment > total) return "Overpay!";
	if(payment < 0) return "Invalid input!";
	if(payment >= 0) return payment;
}

function getBalance(total,payment) {
	let balance = total - payment
	return balance;
}


function editReceipt(total,balance,payment,receipt){
	receipt["Total Paid"] = payment
	receipt["Grand Total"] = total
	receipt["Balance"] = balance
	return receipt;
}

let product = prompt("Enter product name or type 'done' to finish: ");
let sum = 0;
while(product != "done") {
	let price = prompt("Enter price for " + product + " : ");
	sum = getSubTotal(sum,Number(price));	
	invoice = makeInvoice(product,Number(price));
	receipt = makeReceipt(product,Number(price));
	product = prompt("Enter product name or type 'done' to finish: ");
}
if(sum != 0) {
let tax = getVAT(sum)
let total = getTotal(sum,tax)
console.log("-----INVOICE-----")
console.log(editInvoice(sum,tax,total,invoice))
let payment = prompt("Enter payment amount: ")
let check = checkPayment(Number(payment),total)
while(check != Number(payment)) {
	console.log(check)
	payment = prompt("Enter payment amount: ")
	check = checkPayment(Number(payment))
}
let balance = getBalance(total,check)
receipt = editReceipt(total,balance,check,receipt);
console.log("-----RECEIPT-----")
console.log(receipt)
console.log("Payment successful! Thank you for shopping.")
}
else console.log("No purchases made!")

File: Java/DaySix/CheckoutSystem.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Scanner;
public class CheckoutSystem {
public static void main(String[] args) {
Scanner input = new Scanner(System.in);

ArrayList<String[]> invoice = new ArrayList<String[]>();
ArrayList<String[]> receipt = new ArrayList<String[]>();

double sum = 0.0;


System.out.print("Enter product name or type 'done' to finish: ");
String choice = input.nextLine();

while(!choice.equalsIgnoreCase("done")) {

System.out.print("Enter price for " + choice + ": ");
if(input.hasNextDouble() && !choice.trim().isEmpty()) {
	double price = input.nextDouble();
	if(price > 0) {
	String value = String.valueOf(price);
	sum += price;
	invoice.add(CheckoutSystemFunctions.makeInvoice(choice,value));
	receipt.add(CheckoutSystemFunctions.makeReceipt(choice,value));
	System.out.println("Purchase successful!");}
	else 
		System.out.println("Invalid input!");}
else
	System.out.println("Invalid input!");

input.nextLine();
System.out.print("Enter product name or type 'done' to finish: ");
choice = input.nextLine();
}

Double vat = 0.075 * sum;
Double total = sum + vat;
if(sum != 0.0) {
System.out.println("-----INVOICE-----");
for(String[] array : invoice) {
	System.out.printf("%s : %s %n",array[0],array[1]);}
System.out.printf("%s : %.2f \n","Subtotal",sum);
System.out.printf("%s : %.2f \n","VAT(7.5%)",vat);
System.out.printf("%s : %.2f \n","Total Amount",total);}

while(sum != 0.0) {
System.out.print("Enter payment amount: ");
if(input.hasNextDouble()) {
	double payment = input.nextDouble();
	if(payment >= total) {	
		double balance = payment - total;
		System.out.println("-----RECEIPT-----");
		for(String[] array : receipt) {	
			System.out.printf("%s : %s %n", array[0], array[1]);}
		System.out.printf("%s : %.2f %n", "Total Paid",payment);
		System.out.printf("%s : %.2f %n", "Grand Total", total);
		System.out.printf("%s : %.2f %n", "Balance", balance);
		System.out.print("Payment successful! Thank you for shopping!");
		break;
	  }
	if(payment < total && payment > -1) 
		System.out.println("Insufficient!");
	if(payment < 0) 
		System.out.println("Invalid input!");
}
else {
	System.out.println("Invalid input!");
	input.nextLine();
}		
	
}
if(sum == 0.0) System.out.print("No purchases made!");
}
}
